### ISSUE TITLE
Module detection for Automatic-Module-Name fails if META-INF is not 1st or 2nd entry in Jar

### ISSUE THREAD

**Issue by hdecarne** [gradle/gradle#15270]
The current module detection code in org.gradle.internal.jvm.JavaModuleDetector uses the JDK's JarInputStream class to check for the Automatic-Module-Name attribute in the Manifest to determine whether a Jar file can be considered a module or not. The JarInputStream implementation assumes that the META-INF folder is either 1st or 2nd entry in the Jar file. The latter is not true for example for mockito-core-3.6.0.jar. Resulting in not considering this Jar a module (despite it has the proper Automatic-Module-Name attribute set). The following JDK bugs refer to this issue:
* https://bugs.openjdk.java.net/browse/JDK-8031748
* https://bugs.openjdk.java.net/browse/JDK-8211917  

As a result the JarInputStream.getManifest is not a reliable source for the Jar's manifest.

### Expected Behavior
mockito-core Jar is considered a module (named via it's Automatic-Module-Name attribute) and added to module path.

### Current Behavior
mockito-core Jar is not considered a module and not added to the module path. Compilation of module-info.java fails, as the module is not known.

### Context
Now a workaround is needed to force the module detection (as described here https://stackoverflow.com/questions/63204044/gradle-jigsaw-module-not-found) to make the project compile and run with mockito.

### Steps to Reproduce 
See attached project [example.zip](https://github.com/gradle/gradle/files/5579980/example.zip):

```
./gradlew build

> Task :compileJava FAILED
./example/src/main/java/module-info.java:3: error: module not found: org.mockito
        requires org.mockito;
                    ^
1 error

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileJava'.
> Compilation failed; see the compiler error output for details.

```
### Your Environment
```
./gradlew -v

------------------------------------------------------------
Gradle 6.7.1
-
-----------------------------------------------------------

Build time:   2020-11-16 17:09:24 UTC
Revision:     2972ff02f3210d2ceed2f1ea880f026acfbab5c0

Kotlin:       1.3.72
Groovy:       2.5.12
Ant:          Apache Ant(TM) version 1.10.8 compiled on May 10 2020
JVM:          11.0.9.1 (AdoptOpenJDK 11.0.9.1+1)
OS:           Mac OS X 10.16 x86_64

```

Build scan URL: https://gradle.com/s/ddzsnofaxsvoo

**Comment by ljacomet** [comment]
There is no plan to have Gradle do this kind of parsing on its own instead of using the JDK tools for that.
This is also the first library we hear about having that issue. It might be easier to ask the Mockito folks to fix this in their publication.

**Comment by hdecarne** [comment]
Issue has been fixed with mockito-core 3.6.28 (https://github.com/mockito/mockito/pull/2109)
